// next-remove-imports/lib/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _default = function _default() {
  var pluginOptions = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : {};
  return function () {
    var nextConfig = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : {};
    var test = pluginOptions.test || /node_modules([\s\S]*?)\.(tsx|ts|js|mjs|jsx)$/;
    var matchImports = pluginOptions.matchImports || "\\.(less|css|scss|sass|styl)$";
    return Object.assign({}, nextConfig, {
      webpack: function webpack(config, options) {
        if (!options.isServer) {
          config.module.rules.unshift({
            test: test,
            loader: require.resolve('babel-loader', {
              paths: [process.cwd()]
            }),
            options: {
              plugins: [[require.resolve('babel-plugin-transform-remove-imports', {
                // https://nodejs.org/api/modules.html#requireresolverequest-options
                paths: [process.cwd()]
              }), {
                "test": matchImports
              }]]
            }
          });
        }
        if (typeof nextConfig.webpack === 'function') {
          return nextConfig.webpack(config, options);
        }
        return config;
      }
    });
  };
};
exports["default"] = _default;
module.exports = exports.default;
